packages/components/src/timeline/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { __ } from '@wordpress/i18n';
+
+const Timeline = ( { className, items } ) => {
+	const timelineClassName = classnames( 'woocommerce-timeline', className );
+
+	// Sort all the items reverse chronologically
+	items.sort( function( a, b ) {
+		return a.datetime - b.datetime;
+	} );
+
+	// Group the items into local day collections
+	const days = [];
+
+	items.forEach( function( item ) {
+		const itemLocalDatetime = moment.unix( item.datetime );
+		const dayNum = parseInt( itemLocalDatetime.format( 'YYYYMMDD' ), 10 );
+		if ( ! ( dayNum in days ) ) {
+			days[ dayNum ] = [];
+		}
+		days[ dayNum ].push( item );
+	} );
+
+	if ( 0 === items.length ) {
```

Comment:
Could we move this up earlier in the class here to return the "No Events" immediately when we know `items` is empty?

Reply:
Fixed in https://github.com/woocommerce/woocommerce-admin/pull/3614/commits/ce9069e3c8e2835992a279000d78ebc85590ea7d